homeassistant/components/knx/storage/entity_store_schema.py

```diff
@@ -0,0 +1,81 @@
+"""KNX entity store schema."""
+
+import voluptuous as vol
+
+from homeassistant.const import Platform
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.entity import ENTITY_CATEGORIES_SCHEMA
+
+from ..const import SUPPORTED_PLATFORMS_UI
+from ..validation import sync_state_validator
+from .knx_selector import GASelector
+
+BASE_ENTITY_SCHEMA = vol.All(
+    {
+        vol.Optional("name", default=None): vol.Maybe(str),
+        vol.Optional("device_info", default=None): vol.Maybe(str),
+        vol.Optional("entity_category", default=None): vol.Any(
+            ENTITY_CATEGORIES_SCHEMA, vol.SetTo(None)
+        ),
+    },
+    vol.Any(
+        vol.Schema(
+            {
+                vol.Required("name"): str,
+            },
+            extra=vol.ALLOW_EXTRA,
+        ),
+        vol.Schema(
+            {
+                vol.Required("device_info"): str,
+            },
+            extra=vol.ALLOW_EXTRA,
+        ),
+        msg="One of `Device` or `Name` is required",
+    ),
+)
+
+
+SWITCH_SCHEMA = vol.Schema(
+    {
+        vol.Required("entity"): BASE_ENTITY_SCHEMA,
+        vol.Required("knx"): {
+            vol.Optional("invert", default=False): bool,
+            vol.Required("ga_switch"): GASelector(write_required=True),
+            vol.Optional("respond_to_read", default=False): bool,
+            vol.Optional("sync_state", default=True): sync_state_validator,
+        },
+    }
+)
+
+
+ENTITY_STORE_DATA_SCHEMA = vol.All(
+    vol.Schema(
+        {
+            vol.Required("platform"): vol.All(
+                vol.Coerce(Platform),
+                vol.In(SUPPORTED_PLATFORMS_UI),
+            ),
+            vol.Required("data"): dict,
+        },
+        extra=vol.ALLOW_EXTRA,
+    ),
+    cv.key_value_schemas(
+        "platform",
+        {
+            Platform.SWITCH: vol.Schema(
+                {vol.Required("data"): SWITCH_SCHEMA}, extra=vol.ALLOW_EXTRA
+            ),
+        },
+    ),
+)
+
+CREATE_ENTITY_BASE_SCHEMA = {
+    vol.Required("platform"): str,
+    vol.Required("data"): dict,  # validated by ENTITY_STORE_DATA_SCHEMA for platform
+}
+
+UPDATE_ENTITY_BASE_SCHEMA = {
+    vol.Required("entity_id"): str,
+    **CREATE_ENTITY_BASE_SCHEMA,
```

Comment:
Personally I'd use more constants here, because you're also going to reference these fields later on and it avoids typos and stuff. But not a blocker

Reply:
I see your point. 
In development I usually had this and the frontend interface definitions open side by side. It is easier to transfer if no 3rd file with constant declarations has to be looked at.
I'll give it a try with constants.